### PR TITLE
[BEAM-87] Support opt-in for heap dumps

### PR DESCRIPTION
**Pull request by mshields822**

### CODE REVIEW

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/options/DataflowPipelineDebugOptions.java [hunk 1]
```diff
@@ -209,6 +209,19 @@ public Dataflow create(PipelineOptions options) {
...
 
   /**
+   * If {@literal true}, save a heap dump before killing a thread or process which is GC
+   * thrashing or out of memory.
+   *
+   * <p>
+   * CAUTION: Heap dumps can of comparable size to the default boot disk. Consider increasing
+   * the boot disk size before setting this flag to true.
+   */
+  @Description("If {@literal true}, save a heap dump before killing a thread or process "
```
Comment (jkff) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/options/DataflowPipelineDebugOptions.java]: Could you document where the heap dump will be saved?

Comment (mshields822) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/options/DataflowPipelineDebugOptions.java]: That is runner-specific.

Comment (mshields822) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/options/DataflowPipelineDebugOptions.java]: Ah, no it isn't  - was thrown by path (eventually this should go into a runners/dataflow dir in root).

**Comment by mshields822** [comment]
R=klk
Blocking customer impacting bug.

**Comment by mshields822** [comment]
Oh I mean @kennknowles

**Comment by kennknowles** [comment]
Merged.

**Pull request closed without merging**